### DISCUSSION TITLE
How to add unique validation for the translatable column

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#3110] [Help] [answered]
I tried this package [Larave Unique Translation](https://github.com/codezero-be/laravel-unique-translation) but this problem occurred.

```diff
-SQLSTATE[42703]: Undefined column: 7 ERROR: column "data" does not exist
```

[Larave Unique Translation](https://github.com/codezero-be/laravel-unique-translation) tries to take the value of title input but in filament it is data.title.

```php
TextInput::make('title')
    ->unique(
        table: Model::class,
        column: 'title',
        ignorable: fn (?Model $record): ?Model => $record,
        callback: fn (Unique $rule) => UniqueTranslationRule::for((new Model)->getTable())
    )
    ->required(),
```

**Comment by danharrin** [comment, marked as answer]
Since you're passing in a new Rule object to the callback, it will override everything else.

Try this:

```php
->unique(
    callback: fn (?Model $record) => UniqueTranslationRule::for((new Model())->getTable(), 'title')->ignore($record?->getKey())
)
```

**Reply by ArtMin96** [reply]
Thanks @danharrin

**Reply by danharrin** [reply]
Tbh, you can just pass the object to `rule()`, since it is custom:

```php
->rule(UniqueTranslationRule::for()...)
```

**Reply by xdiegom** [reply]
Hi @ArtMin96 , 

Did you have success with this?

It still tries to look for `data.key`. 

 My case is that I want to validate a `slug` uniqueness, so if I tweak (for debug purposes) the `UniqueTranslationValidator` assigning `$attribute = 'slug'` in the beginning of the `validate()` method, the validation fails as expected 🤔.

**Reply by haykaf** [reply]
Hi @xdiegom,

I'm experiencing the same issue with the UniqueTranslationValidator where it's trying to look for data.key. Have you managed to find a solution? If so, could you share it?

**Reply by jackbaron** [reply]
It still not working. Any solution for this?

**Comment by dvandal** [comment]
Fixed for Filament 3.x : https://github.com/codezero-be/laravel-unique-translation/pull/26

**Reply by Nestoran** [reply]
Could you possibly make a fix for Filament 2.x? Or is is the logic that much different there?

**Comment by vivere-dally** [comment]
I'm facing an issue where filament does not send the locale into the validator. Why would that be the case?
